src/sentry/tasks/reports.py

```diff
@@ -436,38 +437,46 @@ def prepare_project_calendar_series(interval, project):
     return clean_calendar_data(project, series, start, stop, rollup)
 
 
-def build_report(name, fields):
-    names, prepare_fields, merge_fields = zip(*fields)
+def build_report(fields):
+    """
+    Constructs the Report namedtuple class, as well as the `prepare` and
+    `merge` functions for creating the Report object.
+
+    Each field is a tuple of the (field name, builder fn, merge fn).
+
+    The merge function is used to merge the value of that field together for
+    multiple reports.
+    """
+    names, field_builders, field_mergers = zip(*fields)
 
-    cls = namedtuple(name, names)
+    cls = namedtuple("Report", names)
```

Comment:
Moved the name into here. I couldn't see a reason that it needed to be part of the call.

Reply:
Personally, I think it should be kept next to where the local is defined so that the local value and the `__name__` are less likely to confusingly diverge.

I think I had originally planned on using this to make different types of reports, hence all the weird meta stuff. Obviously that didn't happen.